File: frontend/src/services/cartService.js
// frontend/src/services/cartService.js
import axios from '@/lib/axios';

const BASE = '/cart';

export const cartService = {
    getCart: () => axios.get(BASE).then(r => r.data.result),

    addItem: (skuId, quantity = 1) =>
        axios.post(`${BASE}/items`, { skuId, quantity }).then(r => r.data.result),

    updateItem: (skuId, quantity) =>
        axios.put(`${BASE}/items`, { skuId, quantity }).then(r => r.data.result),

    removeItem: (skuId) =>
        axios.delete(`${BASE}/items/${skuId}`).then(r => r.data),

    clearCart: () =>
        axios.delete(BASE).then(r => r.data),

    // Guest
    getGuestCart: (sessionId) =>
        axios.get(`${BASE}/guest/${sessionId}`).then(r => r.data.result),

    addToGuestCart: (sessionId, skuId, quantity = 1) =>
        axios.post(`${BASE}/guest/${sessionId}/items`, { skuId, quantity }).then(r => r.data.result),

    // Merge khi login
    mergeCart: (sessionId) =>
        axios.post(`${BASE}/merge?sessionId=${sessionId}`).then(r => r.data.result),
};

File: backend/src/main/java/com/example/clothingstore/service/CartService.java
package com.example.clothingstore.service;

import com.example.clothingstore.dtos.cart.request.CartItemRequest;
import com.example.clothingstore.dtos.cart.response.CartResponse;

public interface CartService {
    CartResponse getCart(String userId);
    CartResponse addItem(String userId, CartItemRequest request);
    CartResponse updateItem(String userId, CartItemRequest request);
    void removeItem(String userId, Long skuId);
    void clearCart(String userId);
    CartResponse mergeGuestCart(String userId, String sessionId);
    CartResponse getGuestCart(String sessionId);
    CartResponse addItemToGuestCart(String sessionId, CartItemRequest request);
}